_posts/2019-02-14-Viewability.md

```diff
@@ -0,0 +1,115 @@
+And viewability measurement and delivery should be the absolute standard nowadays. According to the [definition of the IAB](https://www.iab.com/wp-content/uploads/2015/06/MRC-Viewable-Ad-Impression-Measurement-Guideline.pdf), an advertisement is considered viewable (*crinch*) when more than 50% of its content were in the viewport for more than one second.
+From my point of view, this should be the absolute minimum for an ad to be "delivered".
+But interestingly, the guarantee of viewability is not in high demand. We offer the possibility of only paying for viewable impressions, admittedly, for a markup. But simple math leads you to the inadvertend conclusion, that it is well worth investing the markup if you are interested in viewable impressions. Given, that the average viewability rate in the market is 50%, this should be a no-brainer.
+
+So, now that I got this out of my system, I want to explain, how we measure and deliver viewability.
+
+### Measuring
+
+The viewability itself is measured in the adunit itself. Thus, JavaScript code is executed as part of the AdUnit. This code detects, if we can measure the viewability and if yes, it was viewable. Since we rely on the available APIs, the way of measurement depends strongly on the environment, e.g. mobile web vs. app and which version of APIs and browsers are available.
+
+#### App
+In apps, we first check if a version of [MRAID](https://www.iab.com/guidelines/mobile-rich-media-ad-interface-definitions-mraid/) is available. MRAID is a bridge of the native component of an app into the JavaScript container the adunit is running in. If MRAID is available, we can simply call
+
+```javascript
+mraid.isViewable()
+```
+
+to know, whether the is currently visible or not. But since we need not only to know, if the adunit is viewable, but also how long, we need to monitor the status. Instead of recalling the upper function every couple of milliseconds, we make use viewable change event, specified in MRAID:
+
+```javascript
+var mraidEvent = mraid.EventType ? (mraid.EventType.VIEWABLE_CHANGE || 'viewableChange') : 'viewableChange';
+mraid.addEventListener(mraidEvent, onMraidViewableChange);
+
+function onMraidViewableChange() {
+   mraidViewable = mraid.isViewable();
+}
+```
+
+
+#### Mobile web
+
+The mobile web comes with a bigger variety of environments. Safari and Chrome are the two main browsers and they come with different capabilites. In addition, the available information depends on the fact, if the adunit runs in a friendly or unfriendly iframe. An iframe is unfriendly if it is running with a different source than the original website.
+
+##### 1. Intersection Observer
+The simplest and most reliable case are browsers which support the [IntersectionObserver API](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API). In this case we can just call it with
```

Comment:
suggest: "In this case we can just call it with" -> "In this case we can simply use the following instructions in order to check whether the ad unit is currently viewable."